pkg/plugins/kafka/source.go

```diff
@@ -95,16 +91,21 @@ func (s *Source) startFrom(position record.Position) error {
 	return nil
 }
 
-func toRecord(message *kafka.Message, position string) (record.Record, error) {
-	return record.Record{
+func toRecord(message *kafka.Message, position string) (sdk.Record, error) {
+	return sdk.Record{
 		Position:  []byte(position),
-		CreatedAt: time.Time{},
-		ReadAt:    time.Time{},
-		Key:       record.RawData{Raw: message.Key},
-		Payload:   record.RawData{Raw: message.Value},
+		CreatedAt: message.Time,
+		Key:       sdk.RawData(message.Key),
+		Payload:   sdk.RawData(message.Value),
 	}, nil
 }
 
-func (s *Source) Ack(context.Context, record.Position) error {
+func (s *Source) Ack(ctx context.Context, position sdk.Position) error {
 	return s.Consumer.Ack()
 }
+
+func (s *Source) Teardown(context.Context) error {
+	fmt.Println("Tearing down a Kafka Source...")
+	s.Consumer.Close()
```

Comment:
Teardown can and will be called even if the plugin didn't start consuming, for example when the connector is created Conduit calls `Configure` and then `Teardown`. This line panics in that case.

The other issue is that Conduit logs the error but then happily continues without failing the creation of the connector. Another followup, yay!

Reply:
Followup: https://github.com/ConduitIO/conduit/issues/224

Reply:
👍 Done.